src/dbnode/storage/index.go

```diff
@@ -1136,14 +1188,27 @@ func (i *nsIndex) query(
 }
 
 func (i *nsIndex) execBlockQueryFn(
+	ctx context.Context,
 	cancellable *resource.CancellableLifetime,
 	block index.Block,
 	query index.Query,
 	opts index.QueryOptions,
 	state *asyncQueryExecState,
 	results index.BaseResults,
 ) {
-	blockExhaustive, err := block.Query(cancellable, query, opts, results)
+	ctx, sp := ctx.StartTraceSpan(tracepoint.NSIdxBlockQuery)
```

Comment:
Here too. The only field that’s different from parents is the block start/end time.